src/content/docs/dropins/all/extending.mdx

```diff
@@ -12,50 +12,347 @@ import { Steps } from '@astrojs/starlight/components';
+
+// Extend the cart fragment to include the gift message
+overrideGQLOperations([
+  {
+    // The name of the drop-in to extend
+    npm: '@dropins/storefront-cart',
+    // Additional fields to include in the cart results (gift_message)
+    operations: [
+      `fragment CART_FRAGMENT on Cart {
+        gift_message {
+          from
+          to
+          message
+        }
+      }`
+    ],
+  },
+]);
+```
+
+When you run the install command, the `build.mjs` script will generate a new GraphQL query for the cart drop-in that includes the `gift_message` data.  extend the cart fragment to include the gift message data.
+
+</Task>
+
+<Task>
+
+### Add new data to the payload
+
+Map the updated GraphQL query to the data that the cart events provide to listeners. In this way, the checkout drop-in can access gift message data.
+
+Configure the cart drop-in's initializer to add the new cart data to the existing cart payload. This includes the new data with the payload that the cart events provide to listeners. As an example, here is how it might be configured:
+
+```js title='cart.js'
+/* eslint-disable import/no-cycle */
+import { initializers } from '@dropins/tools/initializer.js';
+import { initialize } from '@dropins/storefront-cart/api.js';
+import { initializeDropin } from './index.js';
+
+initializeDropin(async () => {
+  await initializers.mountImmediately(initialize, {
+    models: {
+      CartModel: {
+        transformer: (data) => {
+          const { gift_message: giftMessage } = data;
+          return {
+            giftMessage,
+          }
+        }
+      }
+    }
+  });
+})();
+```
+
+Now when the cart emits an event with cart data, the `giftMessage` data is included.
+
+</Task>
+
+<Task>
+
+### Retrieve the data and render it
+
+Get the data from the cart event and use it to populate the gift message fields on the checkout page. Here's an example of how you might do this:
+
+```js title='commerce-checkout.js'
+// Event listener to hydrate the new fields with the cart data
+events.on('cart/data', data => {
+  if (!data) return;
+
+  const { id, orderAttributes, giftMessage } = data;
+
+  // Update gift options fields
+  giftOptionsField.setAttribute('cartId', id);
+  if(giftMessage) {
+    giftOptionsField.setAttribute('giftmessage', giftMessage.message);
+    giftOptionsField.setAttribute('fromname', giftMessage.from);
+    giftOptionsField.setAttribute('toname', giftMessage.to);
+  }
+  giftOptionsField.removeAttribute('loading');
+}, { eager: true });
+```
+
+</Task>
+
+<Task>
+
+### Summary
+
+After just a few changes, we were able to add a new feature to the checkout drop-in that allows users to add a gift message to their order. We added a new UI component, integrated the Commerce API to fetch and update gift messages, and extended the data payload for the drop-in to include the gift message data. You can apply these same concepts to any drop-in.
+
+</Task>
+
+</Tasks>
+
+## Extend drop-ins with third-party components
```

Comment:
Consider moving this section  to a separate topic.

Reply:
Yeah, I agree. I'll do that soon on another pass with a separate PR.